src/util/Algorithm.h

```diff
@@ -0,0 +1,25 @@
+// Copyright 2022, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Julian Mundhahs (mundhahj@informatik.uni-freiburg.de)
+
+#pragma once
+
+#include <concepts>
+#include <vector>
+
+namespace ad_utility {
+
+template <class T>
```

Comment:
Use `typename T` instead of `class T` (next thing I'm starting that style guide:)